testslide/lib.py

```diff
@@ -129,6 +129,10 @@ def with_sig_check(*args, **kwargs):
                 )
             if validate_types:
                 _validate_function_signature(callable_template, args, kwargs)
+                if not inspect.iscoroutinefunction(callable_template):
```

Comment:
This will cause issues [when this check fails](https://github.com/cython/cython/issues/2273) I believe.